Sources/AsyncHTTPClient/FileDownloadDelegate.swift

```diff
@@ -23,9 +23,10 @@ public final class FileDownloadDelegate: HTTPClientResponseDelegate {
     public struct Progress: Sendable {
         public var totalBytes: Int?
         public var receivedBytes: Int
+        public var head: HTTPResponseHead!
```

Comment:
Let's avoid the IUO here. I agree with your rationale that it's not a good idea for us to set this to a default value, but I think it's fine to leave it as an explicit optional and just set it to `nil`. This is the same behaviour used for `totalBytes`.

Reply:
@Lukasa

It's just so unfortunate ergonomics-wise because it means user code is going to have to explicitly unwrap (feels bad) or write their own "throw error if head is nil" code even though it will guaranteed to never be `nil` in the `reportProgress` handler or in the result `Response`. I'm almost leaning toward init'ing the head with bogus data now...

Reply:
I went ahead and pushed a change for this that goes down the other proposed route. I strongly believe this will be much cleaner for library users (like me!)

Reply:
Yeah, the cleanliness argument is interesting, but I'm worried about whether we should be enforcing this invariant somewhere else. If we're _that_ confident that we will always have a result, why not store the internal as an optional and project through it before we pass it to the user? We can even have a release-mode fallback that doesn't crash but just provides garbage, and then in debug mode we'd crash.

Reply:
@Lukasa

I'm not familiar with this "project through" pattern in Swift... 

Do you mean store a `var headResponse: HTTPHeadResponse?` optional in `FileDownloadDelegate` and construct a `Progress` struct every time reportProgress is called? Or is there some `Unsafe` pattern I'm not familiar with?

Reply:
I think @Lukasa means something like this:
```swift
private var _head: HTTPResponseHead?
public var head: HTTPResponseHead { _head! }
// or
public var head: HTTPResponseHead { _head ?? .init(...) }
```

Reply:
Yup, that would work nicely.

Reply:
OK, done!